components/SEO.js

```diff
@@ -37,11 +37,42 @@ export const BlogSeo = ({ title, summary, date, lastmod, images = [] }) => {
 
   const featuredImages = imagesArr.map((img) => {
     return {
+      '@type': 'ImageObject',
       url: `${siteMetadata.siteUrl}${img}`,
-      alt: title,
     }
   })
 
+  let authorsArr = authors.length === 0 ? [siteMetadata.author] : authors
+  const authorList = authorsArr.map((author) => {
+    return {
+      '@type': 'Person',
+      name: author,
+    }
+  })
```

Comment:
I did it this way but now JSON-LD pull the name from the frontMatter of the blog post and not the actual name from the authors Markdown files.

Example: For the blog post with the multi-authors, JSON-LD now look like this:

```json
"author": [
    {
      "@type": "Person",
      "name": "default"
    },
    {
      "@type": "Person",
      "name": "sparrowhawk"
    }
  ]
```

As you can see, the results show `'default'` and `'sparrowhawk'` instead of `'Tails Azimuth'` and `'Sparrow Hawk'`.

Reply:
Should have pointed you to the [authorDetails prop](https://github.com/timlrx/tailwind-nextjs-starter-blog/blob/v1/layouts/PostLayout.js#L17) which gets fed into every blog page. Just need to add it in the blogSeo component and the names should be correct.

Reply:
I knew about `authorDetails` and I tried it multiple times yesterday and it didn't work. I kept getting the error message: `TypeError: Cannot read property 'map' of undefined`.

The issue was that I forgot to add `authorDetails={authorDetails}` on [this line](https://github.com/timlrx/tailwind-nextjs-starter-blog/blob/e92b7d2737aae0d9bec2941880f6f96f3ff869b1/layouts/PostLayout.js#L24)

Now, everything should work fine.